reana_server/rest/workflows.py

```diff
@@ -898,6 +898,15 @@ def start_workflow(workflow_id_or_name, user):  # noqa
                 "message": "Status resume is not supported yet."
               }
     """
+
+    def _calculate_complexity(workflow):
+        """Place workflow in queue and calculate and set its complexity."""
```

Comment:
Just a comment: it also silently stores loaded `reana_specification` for Yadage workflows in the database. It could not be obvious for someone later touching the code, so I wonder if it should be mentioned here?

Reply:
Doesn't this happen [here](https://github.com/reanahub/reana-server/pull/367/files#diff-16a8293e6225f903f0412763851a5589f06f6f8891689da7abd0cde7a8166c52R940)? Or you mean the commit?

Reply:
Yes, I mean it was manipulated in one place and committed in another. I guess it's not a big deal, but could be misleading for someone perhaps. I like your solution, it's much cleaner! The only downsize is that it introduces another DB transaction, but I guess it's fine in this case, WDYT?

Reply:
Yes, I'm aware. OTOH, if something goes wrong in the second transaction we will have the reana spec saved.